### PR TITLE
Add test for downloading an empty object

### PR DESCRIPTION
**Pull request by kyleknap**
Related to: https://github.com/boto/s3transfer/pull/65

cc @jamesls @JordonPhillips

### CODE REVIEW

File: tests/integration/customizations/s3/test_plugin.py [hunk 1]
```diff
@@ -497,6 +497,18 @@ def test_download_encrypted_kms_object(self):
         with open(local_filename, 'r') as f:
             self.assertEqual(f.read(), contents)
 
+    def test_download_empty_object(self):
+        bucket_name = self.create_bucket()
```
Comment (jamesls) [inline tests/integration/customizations/s3/test_plugin.py]: We should be able to just use `_SHARED_BUCKET` instead of creating a new bucket.

Comment (kyleknap) [inline tests/integration/customizations/s3/test_plugin.py]: Right forgot about it

**Comment by kyleknap** [comment]
@jamesls updated to use the shared bucket
